### ISSUE TITLE
When running any ML Agent, even from the Examples folder (such as Hallway and Match3), the RuntimeError error occurs: Could not infer dtype of numpy.int32

### ISSUE THREAD

**Issue by Bebebre** [Unity-Technologies/ml-agents#5930]
**Describe the bug**

I have not been able to launch more than one ML-Agent because of this error. It always occurs, even with the use of the --force flag. I think it's because of an incorrect Numpy version, but I never managed to install it below 1.21.0.

**To Reproduce**

1. Install Ml-Agents release 20
2. Install Python 3.10 
3. Install PyTorch 2.0.1
4. Run mlagents-learn --force config\ppo\Hallway.yamal --run-id=1 from the command line
5. See error

**Console logs / stack traces**
I have no idea what it is (

**Screenshots**
![ed](https://github.com/Unity-Technologies/ml-agents/assets/130354848/e3f91541-36a7-4214-82d5-64953f4e2479)
![6](https://github.com/Unity-Technologies/ml-agents/assets/130354848/e3cfca0d-7721-4dbe-80dd-3a9aa2ad6d01)
![r4](https://github.com/Unity-Technologies/ml-agents/assets/130354848/3a028b22-0df6-4ac1-bbd5-429ab0d90575)

**Environment (please complete the following information):**
- Unity Version: [Unity 2021.3.4f1]-
- OS + version: [Windows 10]
- _ML-Agents version_: (0.30.0)
- _Torch version_: (2.0.1)
- _Environment_: (Hallway and Match3)

**NOTE:** If I forgot something, please write about it.

**Comment by Bebebre** [comment]
![ук](https://github.com/Unity-Technologies/ml-agents/assets/130354848/2b4625f0-03f1-4c04-8b41-be067780d9f8)

**Comment by CatRabbitBear** [comment]
I had a similar issue today with the 3DBall example. I got it to work by upgrading numpy to version 1.23.0. I'm not sure what effect this may have on other parts of the framework though.

**Comment by Bebebre** [comment]
Version 1.23.0 doesn't work for me. However, thanks for the idea to double-check the guide, maybe I missed something.

**Comment by miguelalonsojr** [comment]
I noticed that you are using torch 2.0.1. We have not updated ML-Agents yet to use PyTorch 2.x. This error is likely due to incompatibility issues with PyTorch 2.x and other packages like numpy.

**Comment by Bebebre** [comment]
It didn't help, but thanks a lot anyway.
![w5u](https://github.com/Unity-Technologies/ml-agents/assets/130354848/df177c93-e689-4851-8f27-ade202d5ce75)
![3456yt](https://github.com/Unity-Technologies/ml-agents/assets/130354848/43ac43d3-445f-4823-93e1-7450bd31577f)

**Comment by miguelalonsojr** [comment]
Hmmm... very strange. Let me look into this some more. I see that you're also on Windows, which can be troublesome at times. I'll follow up.

**Comment by Bebebre** [comment]
I got desperate and installed ml agents again using python 3.9 and numpy 1.21.1. I also created my project and venv separately, they used to be in the cloned ml-agents repository. Surprisingly, I managed to launch my project, but unfortunately I did not understand the root of this error and the only thing I can advise in such a situation is to reinstall ml-agents again, preferably with python version 3.8.13 to 3.10 and using the "pip install mlagents" command, since it also installs everything the necessary routines.
![w345y](https://github.com/Unity-Technologies/ml-agents/assets/130354848/59e105f9-a214-45de-86db-4b2c7246691c)
